### DISCUSSION TITLE
Transparent background

### DISCUSSION THREAD

**Discussion by propri** [Iron-E/nvim-highlite#28] [Q&A] [answered]
Is it possible to have a transparent background with the highlite colorschemes?

**Comment by Iron-E** [comment, marked as answer]
Absolutely! You can do that by setting the [bg colors](https://github.com/Iron-E/nvim-highlite/blob/master-v4/doc/advanced-usage.md#list-of-palette-colors) of your palette to `"NONE"` where applicable. 

You may also have to adjust the `winblend` setting of individual plugins using their `setup`, and/or manually [override the generated groups](https://github.com/Iron-E/nvim-highlite/blob/master-v4/doc/advanced-usage.md#2a-overriding-groups-optional) to include the `blend` attribute (e.g. `groups.Foo.blend = 25` (see `:h highlight-blend` in neovim). 

For example:

```lua
local Highlite = require 'highlite' --- @type Highlite

local palette, terminal_palette = Highlite.palette 'highlite' -- or derive a palette
palette.bg = 'NONE'
-- you may also want to set `bg_contrast_high` and/or `bg_contrast_low` to 'NONE', depending on your preferred look

local groups = Highlite.groups('highlite', palette)

-- override groups here

Highlite.generate('highlite-custom', groups, terminal_palette)
```

The [complete example](https://github.com/Iron-E/nvim-highlite/blob/master-v4/doc/advanced-usage.md#complete-example) may also help put the above links in context.

Make sure your terminal is configured to have a transparent background as well, or else this will simply make Neovim fall back to your terminal's configured color.

**Reply by propri** [reply]
Thanks. I could easily modify solarized to have a transparent background with this!

```lua
local Highlite = require 'highlite' -- @type Highlite
local palette, terminal_palette = Highlite.palette 'solarized8'
palette.bg = 'NONE'

Highlite.generate('highlite-solarized8-transparent', Highlite.groups('default', palette), terminal_palette)
```
